### PR TITLE
Feature/product service

### PR DESCRIPTION
**Pull request by pprzybys2**

### CODE REVIEW

**Review by wisadr — APPROVED** [review]

File: products/build.gradle [hunk 1]
```diff
@@ -1,8 +1,20 @@
 plugins {
     id 'rush.java-library-conventions'
+    id 'org.springframework.boot' version '2.6.6'
+    id 'io.spring.dependency-management' version '1.0.11.RELEASE'
+    id 'java'
```
Comment (pgancarz) [inline products/build.gradle:5]: after merge with current master this is no longer neccesary

File: products/build.gradle [hunk 2]
```diff
@@ -1,8 +1,20 @@
 plugins {
     id 'rush.java-library-conventions'
```
Comment (pgancarz) [inline products/build.gradle:2]: use spring-service-conventions instead

File: products/build.gradle [hunk 3]
```diff
@@ -1,8 +1,20 @@
...
 dependencies {
-  implementation project(':model')
-  implementation project(':services')
+
+    implementation project(':model')
+    implementation project(':services')
+
+    implementation 'org.springframework.boot:spring-boot-starter'
+    implementation 'net.devh:grpc-server-spring-boot-starter:2.12.0.RELEASE'
+    testImplementation 'org.springframework.boot:spring-boot-starter-test'
```
Comment (pgancarz) [inline products/build.gradle:15]: not necessary any more

File: services/src/main/proto/com/github/rush/services/products_service.proto [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+option java_package = "com.github.rush.services.products";
+
+// Service providing rpcs to find product.
+service ProductsService {
+  rpc FindProduct(ProductRequest) returns (ProductResponse);
+}
+
+message ProductRequest {
+  int32 id = 1;
+  string name = 2;
```
Comment (pgancarz) [inline services/src/main/proto/com/github/rush/services/products_service.proto:15]: let's leave only ID in request. At this moment we don't know which way we are going to request products. We can assume that we will need ID for sure. If we'll need something else then it's way a lot easier to add something to proto than taking care of removing something.

File: services/src/main/proto/com/github/rush/services/products_service.proto [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+option java_multiple_files = true;
+option java_package = "com.github.rush.services.products";
+
+// Service providing rpcs to find product.
+service ProductsService {
+  rpc FindProduct(ProductRequest) returns (ProductResponse);
+}
+
+message ProductRequest {
+  int32 id = 1;
```
Comment (pgancarz) [inline services/src/main/proto/com/github/rush/services/products_service.proto:14]: int64

File: services/src/main/proto/com/github/rush/services/products_service.proto [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+  // NextId = 3;
+}
+
+message ProductResponse {
+  string text = 1;
+  string name = 2;
+  repeated string aliases = 3;
+  repeated string categories = 4;
+  map<int32, string> attributes = 5;
+
```
Comment (pgancarz) [inline services/src/main/proto/com/github/rush/services/products_service.proto:26]: ProductResponse should contain a product field which is of Product type. product should be defined inside model module.

**Review by pgancarz — CHANGES_REQUESTED** [review]

File: services/src/main/proto/com/github/rush/services/products_service.proto (hunk 6 above)
Comment (pprzybys2) [reply services/src/main/proto/com/github/rush/services/products_service.proto:25]: Not sure about that product type. I will move product to model proto later.

**Review by pgancarz — APPROVED** [review]

**Pull request merged**